### PR TITLE
Refactoring of isInWorkingHours method

### PR DESCRIPTION
**Pull request by Ukochka**
# What this PR does
Instead of using string comparing now isBetween method from dayjs is used

## Which issue(s) this PR fixes
https://github.com/grafana/oncall/issues/1418

### CODE REVIEW

File: grafana-plugin/src/components/WorkingHours/WorkingHours.helpers.ts [hunk 1]
```diff
@@ -99,17 +99,30 @@ export const getNonWorkingMoments = (startMoment, endMoment, workingMoments) =>
...
+      .set('date', currentMoment.date())
+      .set('hour', Number(start_HH))
+      .set('minute', Number(start_mm))
+      .set('second', Number(start_ss));
+
+    const endTime = dayjs(momentToStart)
+      .set('date', currentMoment.date())
+      .set('hour', Number(end_HH))
+      .set('minute', Number(end_mm))
+      .set('second', Number(end_ss));
```
Comment (joeyorlando) [inline grafana-plugin/src/components/WorkingHours/WorkingHours.helpers.ts:123]: can this logic which builds the `dayjs` object, be extracted into a separate function? Something like:

```ts
const myFunction = (currentMoment: dayjs.Dayjs, timezone: string, time: string): dayjs.Dayjs => {
  const momentToStart = dayjs().tz(timezone).utcOffset() === 0 ? currentMoment : currentMoment.tz(timezone);

  const [hour, minutes, seconds] = time.split(':');
  return dayjs(momentToStart)
      .set('date', currentMoment.date())
      .set('hour', Number(hour))
      .set('minute', Number(minutes))
      .set('second', Number(seconds));
}
```

then this function can be reused on lines 113 and 119.

**Comment by joeyorlando** [comment]
also, would it make sense to add some unit tests for this function? this way we can be sure it handles all possible cases/edge-cases 🤓

File: grafana-plugin/src/components/WorkingHours/WorkingHours.helpers.ts (hunk 1 above)
Comment (Ukochka) [reply grafana-plugin/src/components/WorkingHours/WorkingHours.helpers.ts]: Done

**Review by maskin25 — CHANGES_REQUESTED** [review]
great that you start using `isBetween` method from dayjs but
working hours can be a list of ranges, in your PR you take into account first range only, you should iterate on full list of ranges
